src/main/java/in/projecteka/consentmanager/consent/ConsentArtefactRepository.java

```diff
@@ -180,6 +175,30 @@
                         }));
     }
 
+    @SneakyThrows
+    public Flux<ConsentExpiry> getConsentArtefacts(ConsentStatus consentStatus) {
+        return Flux.create(fluxSink -> dbClient.preparedQuery(SELECT_CONSENTS_TO_VALIDATE_EXPIRY)
+                .execute(Tuple.of(consentStatus.toString()),
+                        handler -> {
+                            if (handler.failed()) {
+                                fluxSink.error(new Exception("Failed to get GRANTED consents"));
+                            } else {
+                                StreamSupport.stream(handler.result().spliterator(), false)
+                                        .map(row -> {
+                                            JsonObject artefact = (JsonObject) row.getValue("consent_artefact");
+                                            ConsentArtefact consentArtefact = artefact.mapTo(ConsentArtefact.class);
+                                            return ConsentExpiry.builder()
+                                                    .consentId(row.getString("consent_artefact_id"))
+                                                    .patientId(row.getString("patient_id"))
+                                                    .consentExpiryDate(consentArtefact.getPermission().getDataExpiryAt())
+                                                    .build();
+                                        })
+                                        .forEach(fluxSink::next);
+                                fluxSink.complete();
```

Comment:
Use simple for loop as I see there are some issues with Stream in the repo.

Reply:
Sure, Will change